contrib/devtools/lint-whitespace.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+#
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# Check for new lines in diff that introduce trailing whitespace.
+
+# We can't run this check unless we know the commit range for the PR.
+if [ -z "${TRAVIS_COMMIT_RANGE}" ]; then
+  exit 0
+fi
+
+showdiff() {
+  if ! git diff "${TRAVIS_COMMIT_RANGE}" --; then
```

Comment:
I assume you need to set the context lines to 0 to prevent false positives?

`-U0`